### PR TITLE
Support cert-manager for controller tls

### LINKED ISSUE #1461: Integrate with cert-manager to manage secrets on the cluster
**Is your feature request related to a problem? Please describe.**
Currently, the yaml installation, has the caBundle cert for webhooks and private secrets hard-coded in the install.yaml.

**Describe the solution you'd like**
[cert-manager.io](cert-manager.io) provides a mechanism to annotate resources and issue certificates for them while managing the certificates life-cycles.

### PR DESCRIPTION
**Pull request by xxtanisxx**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

To support cert-manager.io in controller TLS similar to allocator service with [CA bundle injector](https://cert-manager.io/docs/concepts/ca-injector/)

Will be introducing the following values:

agones.controller.disableSecret
agones.controller.allocationapiservice.annotations
agones.controller.allocationapiservice.disableCaBundle
agones.controller.validatingwebhook.annotations
agones.controller.validatingwebhook.disableCaBundle
agones.controller.mutatingwebhook.annotations
agones.controller.mutatingwebhook.disableCaBundle

**Which issue(s) this PR fixes**:

Closes #1461

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by xxtanisxx** [comment]
Just signed the CLA

**Comment by xxtanisxx** [comment]
I tried to run above command to install the new changes but it seems like the new arguments is not applied. Am I doing something wrong?

**Comment by markmandel** [comment]
Just gently bumping this PR.

**Comment by xxtanisxx** [comment]
sorry, just realize review happened. I'll update it this afternoon

File: install/helm/agones/templates/extensions.yaml [hunk 1]
```diff
@@ -28,6 +28,10 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- if .Values.agones.controller.webhook.annotations }}
```
[the remark this answers is not in the document]
Comment (xxtanisxx) [reply install/helm/agones/templates/extensions.yaml:31]: sounds good to me. I'll update that.

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 2]
```diff
@@ -282,6 +284,52 @@ For most use cases the controller would have required a restart anyway (eg: cont
...
+  issuerRef:
+    name: selfsigned
+    kind: ClusterIssuer
+EOF
+```
+
+After the certificates are generated, we will want to [inject caBundle](https://cert-manager.io/docs/concepts/ca-injector/) into controller webhook and disable controller secret creation by setting the following:
+
+```bash
+helm install \
```
[the remark this answers is not in the document]
Comment (xxtanisxx) [reply site/content/en/docs/Installation/Install Agones/helm.md:328]: oops, sorry, let me fix that

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 3]
```diff
@@ -120,8 +120,10 @@ The following tables lists the configurable parameters of the Agones chart and t
 | `agones.controller.healthCheck.timeoutSeconds`      | Number of seconds after which the probe times out (in seconds)                                  | `1`                    |
 | `agones.controller.resources`                       | Controller [resource requests/limit][resources]                                                 | `{}`                   |
 | `agones.controller.generateTLS`                     | Set to true to generate TLS certificates or false to provide your own certificates              | `true`                 |
+| `agones.controller.disableSecretCreation`            | Disables the creation of any allocator secrets. If true, you MUST provide the `{agones.releaseName}-cert` secrets before installation. | `false` |
```
[the remark this answers is not in the document]
Comment (xxtanisxx) [reply site/content/en/docs/Installation/Install Agones/helm.md:123]: got it!

File: site/content/en/docs/Installation/Install Agones/helm.md [hunk 4]
```diff
@@ -282,6 +284,52 @@ For most use cases the controller would have required a restart anyway (eg: cont
 You can use our script located at {{< ghlink href="install/helm/agones/certs/cert.sh" >}}cert.sh{{< /ghlink >}} to generate them.
 {{< /alert >}}
 
+Another approach is to use [cert-manager.io](https://cert-manager.io/) solution for cluster level certificate management.
```
[the remark this answers is not in the document]
Comment (xxtanisxx) [reply site/content/en/docs/Installation/Install Agones/helm.md:287]: got it. that make sense

File: install/helm/agones/values.yaml [hunk 5]
```diff
@@ -67,6 +67,9 @@ agones:
     generateTLS: true
     tlsCert: ""
     tlsKey: ""
+    disableSecretCreation: false
+    webhook:
+      annotations: {}
```
Comment (xxtanisxx) [inline install/helm/agones/values.yaml:72]: need to fix this

Comment (xxtanisxx) [reply install/helm/agones/values.yaml]: done

**Review by markmandel — COMMENTED** [review]
Sorry one small formatting change, but other than that - this looks good to go 👍🏻

File: install/helm/agones/templates/extensions.yaml [hunk 6]
```diff
@@ -28,18 +28,24 @@ metadata:
     chart: {{ template "agones.chart" . }}
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
+{{- if .Values.agones.controller.allocationapiservice.annotations }}
```
[the remark this answers is not in the document]
Comment (xxtanisxx) [reply install/helm/agones/templates/extensions.yaml:31]: sounds good. I'll modify all of them including mutatingWebhook and validatingWebhook

**Review by markmandel — APPROVED** [review]
👍🏻👍🏻👍🏻

**Pull request merged**